### PR TITLE
add script parameter to nmap agent

### PR DESCRIPTION
**Pull request by pyzcool**

### CODE REVIEW

File: agent/nmap_options.py [hunk 1]
```diff
@@ -54,11 +55,17 @@ def _set_ports_option(self):
...
 
-
     def _set_timing_option(self):
         """Appends the timing template option to the list of nmap options."""
         return [self.timing_template.value]
 
+    def _set_script(self):
+        """Appends the  option to the list of nmap options."""
+        if self.script is not None:
+            return ['--script', self.script]
```
Comment (amine3) [inline agent/nmap_options.py:65]: --script=<Lua scripts>: <Lua scripts> is a comma separated list of
           directories, script-files or script-categories

You need to persist the file and pass the path

File: ostorlab.yaml [hunk 2]
```diff
@@ -69,3 +69,7 @@ args:
     type: "string"
     description: "Template of timing settings (T0, T1, ... T5)."
     value: "T4"
+  - name: "script"
```
Comment (amine3) [inline ostorlab.yaml:72]: Let's call it scripts and pass a list of strings.
You persist all the scripts in one folder and you pass the folder path to nmap

**Review by amine3 — CHANGES_REQUESTED** [review]

File: ostorlab.yaml (hunk 2 above)
Comment (3asm) [reply ostorlab.yaml:72]: +1

File: agent/nmap_options.py [hunk 3]
```diff
@@ -67,5 +74,5 @@ def command_options(self) -> List[str]:
         command_options.extend(self._set_dns_resolution_option())
         command_options.extend(self._set_ports_option())
         command_options.extend(self._set_timing_option())
+        command_options.extend(self._set_script())
```
Comment (3asm) [inline agent/nmap_options.py:77]: This is all missing unit tests!!!!!

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/nmap_options.py (hunk 3 above)
Comment (pyzcool) [reply agent/nmap_options.py]: when you approve the logic , i will move to unit tests

**Pull request merged**